go-controller/pkg/cluster/ha_master.go

```diff
@@ -0,0 +1,562 @@
+			return err
+		}
+	}
+
+	// Set up leader election process first
+	rl, err := resourcelock.New(
+		resourcelock.ConfigMapsResourceLock,
+		config.Kubernetes.OVNConfigNamespace,
+		haLeaderLockName,
+		hacluster.kubeClient.CoreV1(),
+		nil,
+		resourcelock.ResourceLockConfig{
+			Identity:      hacluster.nodeName,
+			EventRecorder: nil,
+		})
+
+	if err != nil {
+		return err
+	}
+
+	HAClusterNewLeader := func(nodeName string) {
+		logrus.Infof(nodeName + " is the new leader")
+		wasLeader := hacluster.isLeader
+
+		if hacluster.nodeName == nodeName {
+			logrus.Infof(" I (" + hacluster.nodeName + ") won the election. In active mode")
+			err = hacluster.ConfigureAsActive(nodeName)
+			if err != nil {
+				logrus.Errorf(err.Error())
+				if hacluster.manageDBServers {
+					// Stop ovn-northd before panicing.
+					_, _, _ = util.RunOVNNorthAppCtl("exit")
+				}
+				panic(err.Error())
+			}
+			hacluster.isLeader = true
+		} else {
+			logrus.Infof(" I (" + hacluster.nodeName + ") lost the election. In Standby mode")
+			if wasLeader {
+				//This node was leader and it lost the election.
+				// Whenever the node transitions from leader to follower,
+				// we need to handle the transition properly like clearing
+				// the cache. It is better to exit for now.
+				// kube will restart and this will become a follower.
+				if hacluster.manageDBServers {
+					// Stop ovn-northd and then exit.
+					_, _, _ = util.RunOVNNorthAppCtl("exit")
+				}
+				logrus.Infof("I (" + hacluster.nodeName + ") am no longer a leader. Exiting")
+				os.Exit(1)
+			}
+			ep, er := hacluster.cluster.Kube.GetEndpoint(config.Kubernetes.OVNConfigNamespace, ovnkubeDbEp)
+			if er == nil {
+				er = hacluster.ConfigureAsStandby(ep)
+				if er != nil {
+					logrus.Errorf(er.Error())
+					if hacluster.manageDBServers {
+						// Stop ovn-northd and then exit
+						_, _, _ = util.RunOVNNorthAppCtl("exit")
+					}
+					panic(er.Error())
+				}
+			}
+		}
+	}
+
+	lec := leaderelection.LeaderElectionConfig{
+		Lock:          rl,
+		LeaseDuration: leaseDuration,
+		RenewDeadline: renewDeadline,
+		RetryPeriod:   retryPeriod,
+		Callbacks: leaderelection.LeaderCallbacks{
+			OnStartedLeading: func(ctx context.Context) {},
+			OnStoppedLeading: func() {
+				//This node was leader and it lost the election.
+				// Whenever the node transitions from leader to follower,
+				// we need to handle the transition properly like clearing
+				// the cache. It is better to exit for now.
+				// kube will restart and this will become a follower.
+				logrus.Infof("I (" + hacluster.nodeName + ") am no longer a leader. Exiting")
+				if hacluster.manageDBServers {
+					// Stop ovn-northd and then exit.
+					_, _, _ = util.RunOVNNorthAppCtl("exit")
+				}
+				os.Exit(1)
+			},
+			OnNewLeader: HAClusterNewLeader,
+		},
+	}
+
+	//go leaderelection.RunOrDie(context.Background(), lec)
+	le, err := leaderelection.NewLeaderElector(lec)
```

Comment:
`hacluster.leaderElector, err = leaderelection.NewLeaderElector(lec)`

should be possible since 'err' is already defined a ways above this line.

Reply:
Done